### PR TITLE
Button / ButtonGroup documentation

### PR DESCRIPTION
**Pull request by tcbegley**
This PR improves the documentation for `Button` and adds separate documentation for `ButtonGroup`.

It also exposes the `group` prop of `DropdownMenu` so that it can be used inside a `ButtonGroup`, which I realised was an issue while preparing the documentation.

image

image

### CODE REVIEW

**Comment by pbugnion** [comment]
Since we're modifying this documentation, should we also change the description of `n_clicks_timestamp` to suggest using Dash's `callback_context` instead of `n_clicks_timestamp`?

File: src/components/DropdownMenu.js [hunk 1]
```diff
@@ -132,7 +132,12 @@ DropdownMenu.propTypes = {
    * Size of the dropdown. 'sm' corresponds to small, 'md' to medium
    * and 'lg' to large.
    */
-  bs_size: PropTypes.oneOf(['sm', 'md', 'lg'])
+  bs_size: PropTypes.oneOf(['sm', 'md', 'lg']),
+
+  /**
+   * Seet group to True when using inside `ButtonGroup`
```
Comment (pbugnion) [inline src/components/DropdownMenu.js]: Set

Comment (pbugnion) [reply src/components/DropdownMenu.js]: How about 

> Set group to True if the dropdown is inside a `ButtonGroup`.

?

**Comment by pbugnion** [comment]
Looks great!

I made a couple of comments inline, but merge whenever.

File: src/components/DropdownMenu.js (hunk 1 above)
Comment (tcbegley) [reply src/components/DropdownMenu.js]: Sounds good to me

**Pull request merged**